src/Spice86.Core/Emulator/ProgramExecutor.cs

```diff
@@ -175,4 +180,77 @@ private void Dispose(bool disposing) {
             _disposed = true;
         }
     }
+
+    /// <summary>
+    /// Informs the DOS memory manager that the memory where the program was loaded and its stack
+    /// was created is not available to allocate.
+    /// </summary>
+    /// <remarks>
+    /// The stack is allocated right after the executable and grows down. We need to make sure that
+    /// the DOS memory manager doesn't try to allocate that memory when asked via int21h or it will
+    /// overwrite and corrupt parts of the program we're executing or the stack. <br/><br/>
+    /// Arguably DosProcessManager.SetupCpuForExe() would be a better place to do this since it sets
```

Comment:
Couldn't this be in DosMemoryManager? I don't see references to DosProcessManager here (but I am reviewing from a phone which is not very comfortable)